File: src/main/java/com/project/pantau/enums/QueueTab.java
package com.project.pantau.enums;

import java.util.List;

public enum QueueTab {
    OPEN(List.of(ReportStatus.REPORTED, ReportStatus.ACKNOWLEDGED)),
    IN_PROGRESS(List.of(ReportStatus.IN_PROGRESS)),
    RESOLVED(List.of(ReportStatus.RESOLVED));

    private final List<ReportStatus> statuses;

    QueueTab(List<ReportStatus> statuses) {
        this.statuses = statuses;
    }

    public List<ReportStatus> statuses() {
        return statuses;
    }
}

File: internal/enums/queue_tab.go
package enums

type QueueTab string

const (
	QueueTabOpen       QueueTab = "open"
	QueueTabInProgress QueueTab = "in_progress"
	QueueTabResolved   QueueTab = "resolved"
)

func (tab QueueTab) Statuses() []string {
	switch tab {
	case QueueTabOpen:
		return []string{string(ReportStatusReported), string(ReportStatusAcknowledged)}
	case QueueTabInProgress:
		return []string{string(ReportStatusInProgress)}
	case QueueTabResolved:
		return []string{string(ReportStatusResolved)}
	default:
		return nil
	}
}
